### PR TITLE
feat: Added a new getFeatureVariableJson Api

### PR DESCRIPTION
**Pull request by zashraf1985**
## Summary
Implemented getFeatureVariableJson API

## Test plan
Added unit tests for getFeatureVariableJson.

### CODE REVIEW

File: packages/optimizely-sdk/lib/core/optimizely_config/index.tests.js [hunk 1]
```diff
@@ -81,6 +81,9 @@ describe('lib/core/optimizely_config', function() {
         });
         var variablesMap = featuresMap[featureFlag.key].variablesMap;
         featureFlag.variables.forEach(function(variable) {
+          if (variable.type === 'string' && variable.subType === 'json') {
+            variable.type = 'json';
+          }
```
Comment (mjc1283) [inline packages/optimizely-sdk/lib/core/optimizely_config/index.tests.js:86]: Can we add a comment explaining why this is overwritten? Also I think it's a bad idea to mutate `datafile`; it is assigned once at the top of the file and could be used in other tests.

Instead could we do something like this? 
```js
var expectedVariableType =
  variable.type === "string" && variable.subType === "json"
    ? "json"
    : variable.type;
assert.include(variablesMap[variable.key], {
  id: variable.id,
  key: variable.key,
  type: expectedVariableType,
  value: variable.defaultValue,
});
```

File: packages/optimizely-sdk/lib/core/project_config/index.js [hunk 2]
```diff
@@ -119,6 +119,13 @@ export var createProjectConfig = function(datafile) {
 
   projectConfig.featureKeyMap = fns.keyBy(projectConfig.featureFlags || [], 'key');
   objectValues(projectConfig.featureKeyMap || {}).forEach(function(feature) {
+    // Convert type:string and subType:json to type:json to generalize implementation
```
Comment (mjc1283) [inline packages/optimizely-sdk/lib/core/project_config/index.js:122]: I think this comment needs a little more information. We can mention that json variables in the datafile have a `type` of `"string"` for backwards compatibility, but we want to erase the concept of `subType`, and represent them as a first-class type within the project config.

File: packages/optimizely-sdk/lib/optimizely/index.tests.js [hunk 3]
```diff
@@ -3000,6 +3000,36 @@ describe('lib/optimizely', function() {
                   });
                 });
 
+                it('returns the right value from getFeatureVariable and send notification with featureEnabled true', function() {
```
Comment (mjc1283) [inline packages/optimizely-sdk/lib/optimizely/index.tests.js:3003]: These strings passed to `it()` should mention the type of variable being tested (I know you are following what was already here). It's weird that they all have the same description string.

File: packages/optimizely-sdk/lib/core/project_config/index.js (hunk 2 above)
Comment (zashraf1985) [reply packages/optimizely-sdk/lib/core/project_config/index.js]: Done

File: packages/optimizely-sdk/lib/optimizely/index.tests.js (hunk 3 above)
Comment (zashraf1985) [reply packages/optimizely-sdk/lib/optimizely/index.tests.js]: Done

File: packages/optimizely-sdk/lib/core/optimizely_config/index.tests.js (hunk 1 above)
Comment (zashraf1985) [reply packages/optimizely-sdk/lib/core/optimizely_config/index.tests.js]: Done

**Pull request merged**